src/RecyclableMemoryStream.cs

```diff
@@ -540,7 +532,7 @@ public override byte[] GetBuffer()
 
             if (this.blocks.Count > 0 && this.memoryManager.AggressiveBufferReturn)
             {
-                this.memoryManager.ReturnBlocks(this.blocks, this.tag);
+                this.memoryManager.ReturnBlocks(this.blocks, Guid.Empty, this.tag);
```

Comment:
Guid.Empty? This RMS still has an id, why can't we tell that to the RMS manager when we return these blocks?

Reply:
Woops, good catch